framework/src/preconditioners/MoosePreconditioner.C

```diff
@@ -67,15 +70,15 @@ MoosePreconditioner::MoosePreconditioner(const InputParameters & params)
   bool full = getParam<bool>("full");
 
   // We either have a full coupling or a custom coupling
-  if (full && isParamValid("off_diag_row"))
+  if (full && isParamSetByUser("off_diag_row"))
     paramError("off_diag_row", "Set full=false to specify the off-diagonal rows manually");
-  if (full && isParamValid("off_diag_column"))
+  if (full && isParamSetByUser("off_diag_column"))
```

Comment:
I'll stop reviewing here and wait until we have discussed this (we can chat next week). I don't think this file should be modified either.

Reply:
Sure, I can show you how it affects the test cases. Maybe there is better solution for it but simply changing back to `isParamValid` won't work

Reply:
Probably want to check the size instead. That ended up making the most sense in a lot of these changes